### PR TITLE
Added a test for the csvwriter to rapper tests

### PR DESCRIPTION
**Pull request by DLWoodruff**
## Summary/Motivation:
Adds a test for the csvwriter to the rapper tests because users have been claiming
that the csvwriter is failing when they use rapper.

## Changes proposed in this PR:
- adds one test to the rapper tests.
-

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by DLWoodruff** [comment]
All I added was a test, so code coverage going down is an artifact of how coverage is computed and
compared.

File: pyomo/pysp/tests/rapper/rapper_tester.py [hunk 1]
```diff
@@ -110,6 +111,19 @@ def test_ef_solve_with_gap(self):
...
+    def test_ef_solve_with_csvwriter(self):
+        """ solve the ef and report gap"""
+        stsolver = rapper.StochSolver("ReferenceModel.py",
+                                      fsfct = "pysp_instance_creation_callback",
+                                tree_model = self.farmer_concrete_tree)
+        res, gap = stsolver.solve_ef(solvername, tee=True, need_gap=True)
+        csvw.write_csv_soln(stsolver.scenario_tree, "testcref")
+        with open("testcref.csv", 'r') as f:
+            line = f.readline()
+        assert(line.split(",")[0] == "FirstStage")
```
Comment (jsiirola) [inline pyomo/pysp/tests/rapper/rapper_tester.py]: This test needs to delete the file it created:
```python
fname = "testcref.csv"
try:
    csvw.write_csv_soln(stsolver.scenario_tree, fname)
    with open(fname, 'r') as f:
        data = f.readline(s)
finally:
    if os.path.exists(fname):
        os.remove(fname)
```
Other notes:
- This should probably use the tempfile creation logic from either python or pyutilib - we cannot always guarantee that the CWD is writable.
- can we test more than the first entry of the first line?  For example, a sanity check that the remainder of the file is not empty?

**Comment by DLWoodruff** [comment]
@jsiirola 
This test creates a temp directory and cd's to it for all tests, then cd's back at the end. So all files
go in the temp dir.

**Comment by jsiirola** [comment]
@DLWoodruff: Ahh. I didn't catch that from the diff.  That said, we need to add code to the TestSuite `tearDown` to delete the temporary directory.  This (partially) explains why the test server has >32k `/tmp/tmp*` directories on it...

**Pull request merged**